cJSON.c

```diff
@@ -882,10 +889,22 @@ char *cJSON_PrintBuffered(const cJSON *item, int prebuffer, cjbool fmt)
     }
     p.length = prebuffer;
     p.offset = 0;
+    p.noalloc = 0;
 
     return print_value(item, 0, fmt, &p);
 }
 
+int cJSON_PrintPreallocated(cJSON *item,char *buf, const size_t len, const cjbool fmt)
+{
+    char *out;
+    printbuffer p;
+    p.buffer = buf;
+    p.length = len;
+    p.offset = 0;
+    p.noalloc = 1;
+    out = print_value(item,0,fmt,&p);
+    return (out != buf ? -1 : 0);
```

Comment:
Hm. Good point. Considering this, your way of returning true or false might actually be better. An API should be hard to misuse in a dangerous way (like having a double free because of the alias).
The question is how far one is willing to protect the users from themselves.

This will then not allow the use of `cJSON_PrintPreallocated` as an argument to another function (`do_something(cJSON_PrintPreallocated(...))`)

If you do it the way you suggested, you can just `return print_value(...) != NULL;`